### PR TITLE
Use maps:merge_with/3 if available

### PR DESCRIPTION
**Pull request by josefs**
Use OTP 24 in travis and use native implementation of maps:merge_with when available.

### CODE REVIEW

**Comment by josefs** [comment]
Travis doesn't support the latest version of OTP yet. I didn't check before submitting the PR. Oh well, I guess I'll just leave this one here for the time being.

File: src/gradualizer_lib.erl [hunk 1]
```diff
@@ -20,8 +20,11 @@
 %% Number of space used when prettyprinting records for nesting
 -define(PP_RECORD_NESTING_OFFSET, 2).
 
+-if(?OTP_RELEASE >= "24").
+merge_with(F, M1, M2) ->
+    maps:merge_with(F, M1, M2).
```
Comment (zuiderkwast) [inline src/gradualizer_lib.erl:25]: Nice! Do you know who added this to OTP?

**Review by zuiderkwast — COMMENTED** [review]
Looks good. You can merge one of the two commits if you want.

File: src/gradualizer_lib.erl [hunk 2]
```diff
@@ -20,8 +20,11 @@
 %% Number of space used when prettyprinting records for nesting
 -define(PP_RECORD_NESTING_OFFSET, 2).
 
+-if(?OTP_RELEASE >= "24").
```
Comment (zuiderkwast) [inline src/gradualizer_lib.erl:23]: Oops, ?OTP_RELEASE is an integer.

```suggestion
-if(?OTP_RELEASE >= 24).
```

**Review by zuiderkwast — CHANGES_REQUESTED** [review]

**Review by zuiderkwast — APPROVED** [review]

File: src/gradualizer_lib.erl [hunk 3]
```diff
@@ -25,9 +25,11 @@
 %% Number of space used when prettyprinting records for nesting
 -define(PP_RECORD_NESTING_OFFSET, 2).
 
-%% merge_with for maps. Similar to merge_with for dicts.
-%% Arguably, this function should be in OTP.
 -spec merge_with(fun(), map(), map()) -> map().
+-if(?OTP_RELEASE >= 24).
+merge_with(F, M1, M2) ->
+    maps:merge_with(F, M1, M2).
+-else.
```
Comment (zuiderkwast) [inline src/gradualizer_lib.erl:32]: We could have used `-import(maps, [merge_with/3]).` instead. Doesn't matter though.

**Pull request merged**